docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+        emit CommunityGrantEnds();
+    }
+
+    function pauseMarket(bool _paused) external onlyDeployer {
+        require(!contractSealed, "Contract sealed.");
+        marketPaused = _paused;
+    }
+
+    function sealContract() external onlyDeployer {
+        contractSealed = true;
+    }
+
+    //////////////////////////
+    //// ERC 721 and 165  ////
+    //////////////////////////
+
+    function isContract(address _addr) internal view returns (bool addressCheck) {
+        uint256 size;
+        assembly { size := extcodesize(_addr) } // solhint-disable-line
+        addressCheck = size > 0;
+    }
+
+    function supportsInterface(bytes4 _interfaceID) external view override returns (bool) {
+        return supportedInterfaces[_interfaceID];
+    }
+
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes calldata _data) external override {
+        _safeTransferFrom(_from, _to, _tokenId, _data);
+    }
+
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId) external override {
+        _safeTransferFrom(_from, _to, _tokenId, "");
+    }
+
+    function transferFrom(address _from, address _to, uint256 _tokenId) external override canTransfer(_tokenId) validNFToken(_tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == _from, "Wrong from address.");
+        require(_to != address(0), "Cannot send to 0x0.");
+        _transfer(_to, _tokenId);
+    }
+
+    function approve(address _approved, uint256 _tokenId) external override canOperate(_tokenId) validNFToken(_tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(_approved != tokenOwner);
+        idToApproval[_tokenId] = _approved;
+        emit Approval(tokenOwner, _approved, _tokenId);
+    }
+
+    function setApprovalForAll(address _operator, bool _approved) external override {
+        ownerToOperators[msg.sender][_operator] = _approved;
+        emit ApprovalForAll(msg.sender, _operator, _approved);
+    }
+
+    function balanceOf(address _owner) external view override returns (uint256) {
+        require(_owner != address(0));
+        return _getOwnerNFTCount(_owner);
+    }
+
+    function ownerOf(uint256 _tokenId) external view override returns (address _owner) {
+        require(idToOwner[_tokenId] != address(0));
+        _owner = idToOwner[_tokenId];
+    }
+
+    function getApproved(uint256 _tokenId) external view override validNFToken(_tokenId) returns (address) {
+        return idToApproval[_tokenId];
+    }
+
+    function isApprovedForAll(address _owner, address _operator) external override view returns (bool) {
+        return ownerToOperators[_owner][_operator];
+    }
+
+    function _transfer(address _to, uint256 _tokenId) internal {
+        address from = idToOwner[_tokenId];
+        _clearApproval(_tokenId);
+
+        _removeNFToken(from, _tokenId);
+        _addNFToken(_to, _tokenId);
+
+        emit Transfer(from, _to, _tokenId);
+    }
```

Comment:
Do the actual transfer of the NFT. 
Removes existing approvals (for example from opensea).
Emits a Transfer event